vortex-ree/src/compute.rs

```diff
@@ -45,3 +52,39 @@ impl ScalarAtFn for REEArray {
         scalar_at(self.values(), self.find_physical_index(index)?)
     }
 }
+
+impl TakeFn for REEArray {
+    fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        let primitive_indices = flatten_primitive(indices)?;
+        let physical_indices = match_each_integer_ptype!(primitive_indices.ptype(), |$P| {
```

Comment:
I think this should just be a search_sorted? Possibly this? https://github.com/fulcrum-so/vortex/pull/164

Reply:
That said, an iterator would allow us to apply the offset inline without allocating a new buffer

Reply:
This is just search sorted. That’s what find_physical_index does

Reply:
Yeah, but a lot of function calls, vs search_sorted_many

Reply:
Sure